backend/src/services/complectations/complectations.service.ts

```diff
@@ -0,0 +1,225 @@
+import { prisma } from '@data/prisma-client';
+
+import type {
+  ModelReturnedData,
+  ComplectationReturnedData,
+  OptionType,
+  ComplectationsInput,
+  ModelReturnedDbType,
+  ComplectationReturnedDbType,
+} from '@autoline/shared/common/types/types';
+
+const getComplectationsById = async (
+  input: ComplectationsInput,
+): Promise<ModelReturnedData | ComplectationReturnedData> => {
+  const { userId, modelId, complectationId } = input;
+  const data = (await prisma.complectation.findUnique({
+    where: {
+      id: complectationId,
+    },
+    select: {
+      users_wishlists: {
+        where: {
+          user_id: userId,
+        },
+        select: {
+          id: true,
+          complectation: {
+            select: {
+              id: true,
+            },
+          },
+        },
+      },
+      prices_ranges: {
+        select: {
+          price_start: true,
+          price_end: true,
+        },
+      },
+      color: { select: { name: true } },
+      engine_displacement: true,
+      engine_power: true,
+      drivetrain: { select: { name: true } },
+      fuel_type: { select: { name: true } },
+      transmission_type: { select: { name: true } },
+      options: {
+        select: {
+          option: { select: { name: true, type: true } },
+          important: true,
+        },
+      },
+    },
+  })) as ComplectationReturnedDbType;
+
+  const options: OptionType = {
+    security: [],
+    optics: [],
+    multimedia: [],
+    upholstery: [],
+    sound: [],
+    design: [],
+    comfort: [],
+    auxiliary: [],
+    important: [],
+  };
+
+  const importantOptionsList: Set<string> = new Set();
+  data?.options.forEach((option) => {
+    if (option.important === true) {
+      importantOptionsList.add(option.option.name);
+    }
+  }),
+    (options.important = Array.from(importantOptionsList));
```

Comment:
I don't understand what's going on here. You can also get only unique values using the `.filter()`